src/app/pages/MarginTradePage/components/TradingHistory/index.tsx

```diff
@@ -208,16 +211,22 @@ export function TradingHistory() {
       const entries = Object.entries(items);
 
       const closeEntries: CalculatedEvent[] = [];
-      entries.forEach(([, /*loanId*/ events]) => {
+      entries.forEach(([loanId, events]) => {
         // exclude entries that does not have sell events
         if (events.filter(item => item.type === 'sell').length > 0) {
-          const calculation = calculateProfits(events);
-          if (calculation) {
-            closeEntries.push(calculation);
-          }
+          fetch(backendUrl[currentChainId] + '/loanEvents?loanId=' + loanId)
+            .then(response => {
+              return response.json();
+            })
+            .then(loanEvents => {
+              const calculation = calculateProfits(events, loanEvents);
+              if (calculation) {
+                closeEntries.push(calculation);
+              }
+            })
+            .catch(console.error);
```

Comment:
I have hundreds of items in my wallet and takes a minute to fetch data for each of loan.
I think fetching should be removed from here and moved to history row component - so it would fetch only when when component is actually rendered.
So entire row for user would be visible instantly and only profit calculation would loan a little bit